src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int REQUIRED_LENGTH = 6;
+    private static final int REQUIRED_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkForNullOrIncorrectInput(user);
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidInputDataException(
+                    "User with this login - " + user.getLogin()
+                            + " - already registered. Try to log in");
+        }
+        if (validateUserInfo(user)) {
+            storageDao.add(user);
```

Comment:
```suggestion
    @Override
    public User register(User user) {
        checkUserNotNull(user);
        checkUserAge(user);
        checkUserLoginLength(user);
        checkUserPasswordLength(user);
        checkUserNotRegisteredYet(user);
        return storageDao.add(user);
    }
```
you can extract all parameters checks into separate methods and than make the register() like this.
Also redundant "return" as storageDao.add(user) returns the user itself

Reply:
Agree